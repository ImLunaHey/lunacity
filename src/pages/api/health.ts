import type { NextApiRequest, NextApiResponse } from 'next'

const Health = (_request: NextApiRequest, response: NextApiResponse) => {
    response.status(200);
};

export default Health;
