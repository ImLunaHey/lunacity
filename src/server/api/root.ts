import { createTRPCRouter } from './trpc';
import { pageRouter } from './routers/page';
import { tagRouter } from './routers/tag';

/**
 * This is the primary router for your server.
 *
 * All routers added in /api/routers should be manually added here
 */
export const appRouter = createTRPCRouter({
  page: pageRouter,
  tag: tagRouter,
});

// export type definition of API
export type AppRouter = typeof appRouter;
