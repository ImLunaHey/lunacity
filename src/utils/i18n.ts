import i18nNext, { type InitOptions } from 'i18next';
import { initReactI18next } from 'react-i18next';

import Backend, { type HttpBackendOptions } from 'i18next-http-backend';
import LanguageDetector from 'i18next-browser-languagedetector';
import getConfig from 'next/config';
import Pseudo from 'i18next-pseudo';
import { env } from '@app/env.mjs';

const { publicRuntimeConfig } = getConfig() as { publicRuntimeConfig: { APP_URL: string; WS_URL: string; } };
const { APP_URL } = publicRuntimeConfig;

const options = {
    fallbackLng: 'en',
    supportedLngs: ['en', 'ar', 'cn', 'es', 'pt'],
    debug: !!env.NEXT_PUBLIC_LANG_DEBUG,
    defaultNS: 'common',
    returnNull: false,
    load: 'languageOnly' as const,
    react: {
        useSuspense: false,
    },
    ns: ['common'],
    interpolation: {
        escapeValue: false, // not needed for react as it escapes by default
    },
    nonExplicitSupportedLngs: true,
    backend: {
        loadPath: `${APP_URL}/locales/{{lng}}/{{ns}}.json`,
    },
    detection: {
        caches: []
    },
    postProcess: ['pseudo'],
} satisfies InitOptions<HttpBackendOptions>;

export const i18n = i18nNext
    .use(new Pseudo({
        enabled: env.NEXT_PUBLIC_PSUDO_ENABLED === 'true',
        languageToPseudo: 'en-US',
    }))
    // load translation using http -> see /public/locales (i.e. https://github.com/i18next/react-i18next/tree/master/example/react/public/locales)
    // learn more: https://github.com/i18next/i18next-http-backend
    .use(Backend)
    // detect user language
    // learn more: https://github.com/i18next/i18next-browser-languageDetector
    .use(LanguageDetector)
    // pass the i18n instance to react-i18next.
    .use(initReactI18next)
    // init i18next
    // for all options read: https://www.i18next.com/overview/configuration-options
    .init<HttpBackendOptions>(options);
