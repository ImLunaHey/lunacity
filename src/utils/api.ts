/**
 * This is the client-side entrypoint for your tRPC API.
 * It's used to create the `api` object which contains the Next.js App-wrapper
 * as well as your typesafe react-query hooks.
 *
 * We also create a few inference helpers for input and output types
 */
import { createWSClient, httpBatchLink, loggerLink, splitLink, wsLink } from '@trpc/client';
import { createTRPCNext } from '@trpc/next';
import { type inferRouterInputs, type inferRouterOutputs } from '@trpc/server';
import { type NextPageContext } from 'next';
import getConfig from 'next/config';
import superJSON from 'superjson';
import { type AppRouter } from '@app/server/api/root';

const { publicRuntimeConfig } = getConfig() as { publicRuntimeConfig: { APP_URL: string; WS_URL: string; } };
const { APP_URL, WS_URL } = publicRuntimeConfig;
const getEndingLink = (ctx: NextPageContext | undefined) => {
  if (typeof window === 'undefined') {
    return httpBatchLink({
      url: `${APP_URL}/api/trpc`,
      headers() {
        if (ctx?.req) {
          // on ssr, forward client's headers to the server
          return {
            ...ctx.req.headers,
            'x-ssr': '1',
          };
        }
        return {};
      },
    });
  }

  // Make Websocket client
  const client = createWSClient({
    url: WS_URL,
  });

  // If we're on the client create a split link that sends both queries
  // and mutations over http-batch and all subscriptions over websocket
  return splitLink({
    condition(op) {
      return op.type !== 'subscription';
    },
    // HTTP batching
    true: httpBatchLink({
      url: `${APP_URL}/api/trpc`,
      headers() {
        if (ctx?.req) {
          // on ssr, forward client's headers to the server
          return {
            ...ctx.req.headers,
            'x-ssr': '1',
          };
        }
        return {};
      },
    }),
    // Websocket
    false: wsLink<AppRouter>({
      client,
    })
  });
}

/**
 * A set of typesafe react-query hooks for your tRPC API
 */
export const api = createTRPCNext<AppRouter>({
  config({ ctx }) {
    return {
      /**
       * Transformer used for data de-serialization from the server
       * @see https://trpc.io/docs/data-transformers
       **/
      transformer: superJSON,

      /**
       * Links used to determine request flow from client to server
       * @see https://trpc.io/docs/links
       * */
      links: [
        loggerLink({
          enabled: (opts) =>
            process.env.NODE_ENV === 'development' ||
            (opts.direction === 'down' && opts.result instanceof Error),
        }),
        getEndingLink(ctx)
      ]
    };
  },
  /**
   * Whether tRPC should await queries when server rendering pages
   * @see https://trpc.io/docs/nextjs#ssr-boolean-default-false
   */
  ssr: false,
});

/**
 * Inference helper for inputs
 * @example type HelloInput = RouterInputs['example']['hello']
 **/
export type RouterInputs = inferRouterInputs<AppRouter>;
/**
 * Inference helper for outputs
 * @example type HelloOutput = RouterOutputs['example']['hello']
 **/
export type RouterOutputs = inferRouterOutputs<AppRouter>;
