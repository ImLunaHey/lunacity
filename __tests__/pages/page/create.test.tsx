import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Create from '@app/pages/page/create';
import { api } from '@app/utils/api';
import { replaceDynamicIds } from '__tests__/__utils__/replace-dynamic-ids';

jest.mock('@app/utils/api', () => ({
  api: {
    page: {
      createPage: {
        useMutation: jest.fn(() => ({
          mutate: jest.fn(),
        })),
      },
    },
  },
}));

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({
    asPath: '/',
  })),
}));

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn(() => ({
    t: jest.fn((text: string) => text),
  })),
}));

describe('Create', () => {
  it('renders a create page form', () => {
    const { container } = render(<Create />);
    expect(replaceDynamicIds(container)).toMatchSnapshot();

    expect(api.page.createPage.useMutation).toBeCalledTimes(1);
  });
});
