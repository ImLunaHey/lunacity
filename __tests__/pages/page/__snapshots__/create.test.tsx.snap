// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create renders a create page form 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center"
  >
    <div
      class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-ilimyxf-css"
      role="section"
      tabindex="-1"
    >
      <form>
        <p
          class="nextui-c-PJLV nextui-c-PJLV-hyvuql-weight-bold nextui-c-PJLV-ibsXxfh-css"
        >
          Create a new page
        </p>
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-ifGHEql-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-dOamyD-bordered-true nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-c-eXOOPO-eNajNl-cv nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--bordered nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-gieWTI nextui-c-gieWTI-ceZkzL-isRight-false nextui-c-gieWTI-cMnJPo-bordered-true nextui-c-gieWTI-gQKSQY-cv nextui-input-label--left"
              >
                @
              </span>
              <input
                aria-label="Handle"
                aria-multiline="false"
                aria-placeholder=""
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-c-jeuecp-bXqash-bordered-true nextui-input"
                data-state="normal"
                id="static-id-0"
                maxlength="25"
                name="handle"
                placeholder=""
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-iijuxSl-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-ifGHEql-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-dOamyD-bordered-true nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-c-eXOOPO-eNajNl-cv nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--bordered nextui-input-wrapper--shadow"
            >
              <input
                aria-label="Display name"
                aria-multiline="false"
                aria-placeholder=""
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-c-jeuecp-bXqash-bordered-true nextui-input"
                data-state="normal"
                id="static-id-1"
                maxlength="50"
                name="displayName"
                placeholder=""
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-c-gJvMas-bDGmTT-withValue-true nextui-input-helper-text-container"
          >
            <p
              class="nextui-c-cwdhZi nextui-input-helper-text"
            >
              The display name
            </p>
          </div>
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-iijuxSl-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-ifGHEql-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-iNKybL-animated-true nextui-input-container nextui-input-container--normal nextui-input-container--textarea"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-dOamyD-bordered-true nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-ccrOPm-underlined-false nextui-c-eXOOPO-iLMmYc-isTextarea-true nextui-c-eXOOPO-eNajNl-cv nextui-c-eXOOPO-dpaosk-cv nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--bordered nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                Add your socials here?
              </span>
              <textarea
                aria-label="Page description"
                aria-multiline="true"
                aria-placeholder="Add your socials here?"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-c-jeuecp-dYLnuG-isTextarea-true nextui-c-jeuecp-bXqash-bordered-true nextui-input-textarea"
                data-state="normal"
                id="static-id-2"
                maxlength="100000"
                name="description"
                placeholder="Add your socials here?"
                type="text"
              />
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-c-gJvMas-bDGmTT-withValue-true nextui-input-helper-text-container"
          >
            <p
              class="nextui-c-cwdhZi nextui-input-helper-text"
            >
              The page description
            </p>
          </div>
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-iijuxSl-css"
        />
        <button
          class="nextui-c-iWjDFM nextui-c-gulvcB nextui-c-iWjDFM-hkKLfu-color-default nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready min-w-full"
          data-state="ready"
          type="submit"
        >
          <span
            class="nextui-button-text"
          >
            Create Page
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
